src/execution_plan/ops/op_unwind.c

```diff
@@ -73,8 +87,18 @@ static OpResult UnwindInit(OpBase *opBase) {
  * NULL will be returned if dynamic list is not evaluted (listIdx = INDEX_NOT_SET)
  * or in case where the current list is fully consumed. */
 Record _handoff(OpUnwind *op) {
-	// If there is a new value ready, return it.
-	if(op->listIdx < SIArray_Length(op->list)) {
+	if(op->is_range && !SIValue_IsNull(op->current)) {
+		if(SIValue_Compare(op->current, op->to, NULL) != 0) {
+			Record r = OpBase_CloneRecord(op->currentRecord);
```

Comment:
Seems like `Record r = OpBase_CloneRecord(op->currentRecord);` is called rather or not this condition `if(SIValue_Compare(op->current, op->to, NULL) != 0)` is true, consider moving it outside the block